src/src/main/java/controller/ProductRegistrationServlet.java

```diff
@@ -0,0 +1,43 @@
+package controller;
+
+import dao.impl.ProductDAO;
+import factories.ProductDAOFactory;
+import model.Product;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/new_product")
+public class ProductRegistrationServlet extends HttpServlet {
+
+    private static final ProductDAO PRODUCT_DAO = ProductDAOFactory.getInstance();
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        request.getServletContext().getRequestDispatcher("/new_product.jsp").forward(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        String name = request.getParameter("name");
+        String description = request.getParameter("description");
+        double price = -1;
```

Comment:
what the magic number is it?

Reply:
This variable is used to check that the value of the price field is greater than 0. In classes ProductDAO and UsersDAO overridden method add created by the implementation of the interface and if I change it to User or Product it gives me an error.